src/StatusBar.ts

```diff
@@ -1,26 +1,62 @@
-import { window, StatusBarAlignment } from 'vscode'
+import * as vscode from 'vscode'
 import * as elegantSpinner from 'elegant-spinner'
+import { extensionName } from './appGlobals'
+import { JestExt } from './JestExt'
+
+enum StatusType {
+  active,
+  summary,
+}
 
 type Status = 'running' | 'failed' | 'success' | 'stopped' | 'initial'
-type QueueItem = {
+interface QueueItem {
   source: string
   status: Status
   details: string | undefined
 }
 
+interface StatusBarSpinner {

```

Comment:
Not the best name for such composite entity. I'd prefer to have two separate spinners, like `StatusItem` below

Reply:
I agree. Actually, I think there are simply too many 'get x by statusType' kind of logic throughout the code, time to do some refactoring to consolidate those into a self-contained block...